plugins/resin/src/main/java/com/navercorp/pinpoint/plugin/resin/ResinPlugin.java

```diff
@@ -149,53 +130,46 @@ private void addServletInvocation() {
 				if (serviceMethodEditorBuilder != null) {
 					serviceMethodEditorBuilder.addInterceptor("com.navercorp.pinpoint.plugin.resin.interceptor.ServletInvocationInterceptor");
```

Comment:
You could change this to
```
serviceMethodEditorBuilder.addScopedInterceptor("com.navercorp.pinpoint.plugin.resin.interceptor.ServletInvocationInterceptor", $SCOPE_NAME, ExecutionPolicy.BOUNDARY);
```
so that you don't have to do the `checkRequestDispatcherImpl()` check inside `ServletInvocationInterceptor`.

Scoped interceptors with `ExecutionPolicy.BOUNDARY` will only run when they're not inside the same scope (identified by $SCOPE_NAME above). So in this case, `ServletInvocationInterceptor` will only run once at the beginning of a request, and won't run when it's invoked by `RequestDispatcherImpl` as it's inside the scope already.